### PR TITLE
Add sequence_reverse_op

### PR DESCRIPTION
**Pull request by sneaxiy**

### CODE REVIEW

File: paddle/fluid/operators/sequence_reverse_op.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+class SequenceReverseOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "The input LoDTensor of sequence_reverse op.");
+    AddOutput("Y", "The output LoDTensor of sequence_reverse op.");
+    AddComment(R"DOC(
+SequenceReverse Operator
+
+Reverse the sequence according to LoD.
```
Comment (panyx0718) [inline paddle/fluid/operators/sequence_reverse_op.h]: Please add examples here.

Comment (sneaxiy) [reply paddle/fluid/operators/sequence_reverse_op.h]: Done.

**Review by panyx0718 — DISMISSED** [review]
LG API

**Review by panyx0718 — DISMISSED** [review]

**Review by typhoonzero — DISMISSED** [review]
LGAPI

File: paddle/fluid/operators/sequence_reverse_op.h [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput("X"), "Input(X) must exist");
+    PADDLE_ENFORCE(ctx->HasOutput("Y"), "Output(Y) must exist");
+
+    auto x_dim = ctx->GetInputDim("X");
+    PADDLE_ENFORCE_GE(x_dim.size(), 2,
+                      "Rank of Input(X) must be not less than 2.");
+
+    ctx->SetOutputDim("Y", x_dim);
```
Comment (chengduoZH) [inline paddle/fluid/operators/sequence_reverse_op.h:36]: You should use `ShareDim` here.

File: paddle/fluid/operators/sequence_reverse_op.cc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/operators/sequence_reverse_op.h"
+
+namespace ops = paddle::operators;
+
+REGISTER_OPERATOR(sequence_reverse, ops::SequenceReverseOp,
+                  ops::SequenceReverseOpMaker,
+                  ops::SequenceReverseGradOpDescMaker);
```
Comment (chengduoZH) [inline paddle/fluid/operators/sequence_reverse_op.cc:21]: You had better add `SequenceReverseInferVarType`.

**Review by chengduoZH — DISMISSED** [review]

**Review by typhoonzero — APPROVED** [review]
LGAPI

File: paddle/fluid/operators/sequence_reverse_op.cc (hunk 3 above)
Comment (panyx0718) [reply paddle/fluid/operators/sequence_reverse_op.cc:21]: @sneaxiy seems better to add one in follow up.

File: paddle/fluid/operators/sequence_reverse_op.h (hunk 2 above)
Comment (panyx0718) [reply paddle/fluid/operators/sequence_reverse_op.h:36]: What's the consequence of using `SetOutputDim` instead of `ShareDim`? @chengduoZH

**Review by panyx0718 — APPROVED** [review]
LG API

File: paddle/fluid/operators/sequence_reverse_op.h (hunk 2 above)
Comment (chengduoZH) [reply paddle/fluid/operators/sequence_reverse_op.h:36]: If the input(`X`) is SelecetedRows,  `ctx->GetInputDim("X")` will return the completed dim which replaced the 0th with `height`. But `SequenceReverse`'s input and output are `LoDTensor` here, so,  it is ok to use `SetOutputDim ` here.

**Pull request merged**